.github/workflows/cleanup.yml

```diff
@@ -0,0 +1,43 @@
+# Copyright 2023 The Authors (see AUTHORS file)
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#      http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Workflow that handles AOD requests cleanup.
+name: 'aod-cleanup'
+
+on:
+  pull_request:
+    types: 'closed'
+    branches:
+      - 'main'
+    paths:
+      - 'tool.yaml'
+
+concurrency:
+  group: '${{ github.workflow }}-${{ github.head_ref || github.ref }}-aod-cleanup'
+  cancel-in-progress: true
+
+jobs:
+  validate:
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+      pull-requests: 'write'
+    uses: 'abcxyz/access-on-demand/.github/workflows/tool_cleanup.yml@2eba0549e4a5879ffcfa129d5fcdfcf10985d0d2' # Change here to pin the tag or commit.
```

Comment:
Should rename the reusable workflow as well?

Reply:
Can I rename the workflow later when it is updated with IAM cleanup? I need to update the current tool_cleanup.yaml to do IAM cleanup, I will rename the workflow as well, and then here.

Reply:
Sure. You want to get this in first or wait?

Reply:
Check in this one first, I also want to check the approved check.